suite-native/screen-overlay/src/components/ScreenshotProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, MutableRefObject, useCallback, useMemo, useState } from 'react';
+import { View } from 'react-native';
+
+import { makeImageFromView, SkImage } from '@shopify/react-native-skia';
+
+const SNAPSHOT_TIMEOUT = 1000;
+
+export const ScreenshotContext = createContext<{
+    screenshot: SkImage | null;
+    takeScreenshot: (screenViewRef: MutableRefObject<View | null>) => void;
+}>({ screenshot: null, takeScreenshot: () => {} });
+
+export const ScreenshotProvider = ({ children }: { children: React.ReactNode }) => {
+    const [screenshot, setScreenshot] = useState<SkImage | null>(null);
+
+    const takeScreenshot = useCallback((screenViewRef: MutableRefObject<View | null>) => {
+        if (!screenViewRef) return;
+
+        // The timeout is needed because sometimes happens that the target view of the snapshot is already defined
+        // but the UI is not rendered yet. If you try to take snapshot in such situation, the Skia crashes the app.
+        // see more: https://github.com/Shopify/react-native-skia/discussions/1648
+        setTimeout(async () => {
+            const freshSnapshot = await makeImageFromView(screenViewRef);
```

Comment:
Comment from attached issue. Can we please also use `onLayout` instead of timeout? It's unreliable and could still result in app crashes in edge cases.

Reply:
I was trying to use this approach. But the problem is that there is no View, that would comply with these two conditions:
1. to be high enough in the tree that it would contain the whole screen even with the bottom navigation tab bar 
2. re-render on every navigation state change.

For this reason, I solved that by implementing a react-navigation event listener which takes a screenshot on every navigation state change.